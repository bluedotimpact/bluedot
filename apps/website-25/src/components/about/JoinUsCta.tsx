import { CTALinkOrButton } from '@bluedot/ui';

const JoinUsCta = () => {
  return (
    <div className="join-us-cta mx-16 my-8 max-w-full overflow-hidden">
      <div className="join-us-cta__container container-lined flex flex-col justify-center items-center py-20 bg-[radial-gradient(ellipse_at_30%,_#fff_0,_#6687ff_100%)]">
        <h3 className="join-us-cta__title max-w-[890px] text-center mb-8">Join us in our mission to ensure humanity safely navigates the transition to transformative AI.</h3>
        <CTALinkOrButton
          className="join-us-cta__cta-button"
          variant="primary"
          withChevron
          url="/careers"
        >
          View our careers
        </CTALinkOrButton>
      </div>
    </div>
  );
};

export default JoinUsCta;
