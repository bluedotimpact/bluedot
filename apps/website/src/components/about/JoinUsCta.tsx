import { CTALinkOrButton } from '@bluedot/ui';
import { H3 } from '../Text';
import { ROUTES } from '../../lib/routes';

const JoinUsCta = () => {
  return (
    <div className="join-us-cta w-full max-w-max-width mx-auto lg:px-spacing-x my-spacing-y overflow-hidden">
      <div className="join-us-cta__container w-full lg:container-lined flex flex-col justify-center items-center py-20 bg-[radial-gradient(ellipse_at_30%,_#fff_0,_#6687ff_100%)]">
        <H3 className="join-us-cta__title max-w-[890px] text-center mb-8">Join us in our mission to ensure humanity safely navigates the transition to transformative AI.</H3>
        <CTALinkOrButton
          className="join-us-cta__cta-button"
          variant="primary"
          withChevron
          url={ROUTES.joinUs.url}
        >
          View our careers
        </CTALinkOrButton>
      </div>
    </div>
  );
};

export default JoinUsCta;
