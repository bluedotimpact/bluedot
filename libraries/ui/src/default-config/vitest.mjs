/* eslint-disable import/no-extraneous-dependencies */
import { defineConfig, defaultExclude } from 'vitest/config';
import react from '@vitejs/plugin-react';
import * as dotenv from 'dotenv';

/**
 * @param {import('vitest/config').UserConfig} [config] 
 * @returns {import('vitest/config').UserConfig}
 */
export const withDefaultBlueDotVitestConfig = async (config) => defineConfig({
  ...config,
  plugins: [react(), ...(config?.plugins ?? [])],
  test: {
    environment: 'happy-dom',
    env: dotenv.config({ path: '.env.test' }).parsed,

    ...config?.test,

    exclude: [...defaultExclude, '**/.{turbo,next}/**', ...(config?.test?.exclude ?? [])],
  },
});
