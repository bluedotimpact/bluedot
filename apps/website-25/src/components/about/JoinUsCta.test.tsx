import { render } from '@testing-library/react';
import { describe, expect, test } from 'vitest';
import JoinUsCta from './JoinUsCta';

describe('JoinUsCta', () => {
  test('renders default as expected', () => {
    const { container } = render(<JoinUsCta />);
    expect(container).toMatchSnapshot();
  });
});
