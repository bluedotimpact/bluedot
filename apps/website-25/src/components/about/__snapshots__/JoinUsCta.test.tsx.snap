// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`JoinUsCta > renders default as expected 1`] = `
<div>
  <div
    class="join-us-cta my-8 lg:mx-16 overflow-hidden"
  >
    <div
      class="join-us-cta__container w-full max-w-max-width lg:container-lined flex flex-col justify-center items-center py-20 bg-[radial-gradient(ellipse_at_30%,_#fff_0,_#6687ff_100%)]"
    >
      <h3
        class="join-us-cta__title max-w-[890px] text-center mb-8"
      >
        Join us in our mission to ensure humanity safely navigates the transition to transformative AI.
      </h3>
      <a
        class="cta-button flex items-center justify-center rounded-radius-sm transition-all duration-200 text-sm px-4 py-3 w-fit font-[650] whitespace-nowrap cta-button--primary bg-bluedot-normal link-on-dark join-us-cta__cta-button"
        data-testid="cta-link"
        href="/careers"
      >
        <span
          class="cta-button__text"
        >
          View our careers
        </span>
        <span
          class="cta-button__chevron ml-3"
        >
          <img
            alt="â†’"
            class="cta-button__chevron-icon size-2"
            src="/icons/chevron_white.svg"
          />
        </span>
      </a>
    </div>
  </div>
</div>
`;
